### PR TITLE
feat: [NOD-150] API-CFG Alarms

### PR DESCRIPTION
**Pull request by aomegax**
### List of changes
- added JDBC connection problem to API-Config
- added Out-of-memory problem to API-Config-Cache
- moved DB-Check alarm to 00_alarm.tf

### Type of changes

- [x] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/apiconfig-app/00_alert_apiconfig_cache.tf [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+## API-CONFIG CACHE ##
+######################
+
+locals {
+
+  oracle_version     = "oracle"
+  postgresql_version = "postgresql"
+}
+
+# Pod OutOfMemory - ORACLE
```
Comment (pasqualespica) [inline src/domains/apiconfig-app/00_alert_apiconfig_cache.tf:11]: can you avoid c&p alert code using [for_each](https://github.com/pagopa/pagopa-infra/blob/6c1e5b4d95c2952a3e4d55208e5f9d7251f27379/src/domains/nodo-app/00_alert.tf#L163) ?

File: src/domains/apiconfig-app/00_alert_apiconfig_core.tf [hunk 2]
```diff
@@ -0,0 +1,74 @@
+#####################
+## API-CONFIG CORE ##
+#####################
+
+# Node database availability
```
Comment (pasqualespica) [inline src/domains/apiconfig-app/00_alert_apiconfig_core.tf:5]: why `Node database availability` aren't param for orc/pg ?

File: src/domains/apiconfig-app/00_alert_apiconfig_core.tf [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  time_window = 45
+  trigger {
+    operator  = "GreaterThanOrEqual"
+    threshold = 3
+  }
+}
+
+# JDBC connection problem
+resource "azurerm_monitor_scheduled_query_rules_alert" "apiconfig_jdbc_connection" {
+  name                = format("%s-%s", module.api_config_app_service.name, "jdbc-connection")
```
Comment (pasqualespica) [inline src/domains/apiconfig-app/00_alert_apiconfig_core.tf:40]: ditto why not param for orc/pg ?

File: src/domains/apiconfig-app/00_alert_apiconfig_core.tf [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    action_group           = [data.azurerm_monitor_action_group.email.id, data.azurerm_monitor_action_group.slack.id]
+    email_subject          = "JDBC Connection"
+    custom_webhook_payload = "{}"
+  }
+  data_source_id = data.azurerm_application_insights.application_insights.id
+  description    = "Problem to acquire JDBC connection"
+  enabled        = true
+
+  query = format(<<-QUERY
+    let threshold = 0.05;
```
Comment (pasqualespica) [inline src/domains/apiconfig-app/00_alert_apiconfig_core.tf:54]: could be better using threshold as % , 5 instead of 0.05 ?

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: src/domains/apiconfig-app/00_alert_apiconfig_core.tf (hunk 4 above)
Comment (aomegax) [reply src/domains/apiconfig-app/00_alert_apiconfig_core.tf]: done

File: src/domains/apiconfig-app/00_alert_apiconfig_core.tf (hunk 2 above)
Comment (aomegax) [reply src/domains/apiconfig-app/00_alert_apiconfig_core.tf:5]: pg is a next PR

File: src/domains/apiconfig-app/00_alert_apiconfig_core.tf (hunk 3 above)
Comment (aomegax) [reply src/domains/apiconfig-app/00_alert_apiconfig_core.tf:40]: pg is a next PR

File: src/domains/apiconfig-app/00_alert_apiconfig_cache.tf (hunk 1 above)
Comment (aomegax) [reply src/domains/apiconfig-app/00_alert_apiconfig_cache.tf]: done

**Review by pasqualespica — APPROVED** [review]
🔝

**Pull request merged**